File: src/rapidmq/message_broker.py
class MessageBroker:
    # ... existing code ...

    def handle_edge_node_sync(self, edge_node_id, messages):
        for msg_id, topic, payload, timestamp in messages:
            self.publish(topic, payload, timestamp)
        return [msg[0] for msg in messages]  # Return list of processed message IDs

    # ... existing code ...

File: src/message_broker.rs
use serde_json::Value;

pub struct MessageBroker {
    // ... existing fields ...
}

impl MessageBroker {
    // ... existing methods ...

    pub fn handle_edge_node_sync(&self, edge_node_id: &str, messages: Vec<(i64, String, String, i64)>) -> Vec<i64> {
        let mut processed_ids = Vec::new();
        for (msg_id, topic, payload, timestamp) in messages {
            if let Ok(payload_value) = serde_json::from_str::<Value>(&payload) {
                self.publish(&topic, payload_value, timestamp);
                processed_ids.push(msg_id);
            }
        }
        processed_ids
    }

    fn publish(&self, topic: &str, payload: Value, timestamp: i64) {
        // Implement the logic to publish the message to subscribers
        // This would involve updating internal data structures and notifying subscribers
    }
}